.github/workflows/generate-static-assets.yml

```diff
@@ -0,0 +1,68 @@
+name: Generate Static Assets
+
+on:
+  push:
+    branches:
+      - crt/1.2.x
+env:
+  GO_TAGS: "release"
+
+jobs:
+  get-go-version:
+    runs-on: ubuntu-latest
+    outputs:
+      go-version: ${{ steps.get-go-version.outputs.go-version }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Determine Go version
+        id: get-go-version
+        # We use .go-version as our source of truth for current Go
+        # version, because "goenv" can react to it automatically.
+        run: |
+          echo "Building with Go $(cat .go-version)"
+          echo "::set-output name=go-version::$(cat .go-version)"
+  generate-metadata:
+    runs-on: ubuntu-latest
+    needs: get-go-version
+    name: generate-metadata
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ needs.get-go-version.outputs.go-version }}
+
+      - name: Setup node and yarn
+        uses: actions/setup-node@v2
+        with:
+          node-version: "14"
+          cache-dependency-path: "ui/yarn.lock"
+      - name: Install Yarn
+        run: |
+          npm install -g yarn
+      - uses: actions/checkout@v2
+      - name: Generate static assets
+        id: generate-static-assets
+        run: |
+          make deps
+          make prerelease
+      - name: Commit and push changes
+        id: commit-change-push
+        run: |
+          git add -A . 
+          find . -name '*.generated.go' -not -path './vendor/*' -exec git add -f '{}' \;
+          if ! git diff-index --quiet HEAD --;
+          then
+            git config --global user.email "github-team-nomad-core@hashicorp.com"
+            git config --global user.name "hc-github-team-nomad-core"
+            git commit --message "Generate files for $(NOMAD_VERSION) release"
+            git push origin "$(git rev-parse --abbrev-ref HEAD)"
+            echo "committing generated files"
+          else
+            echo "no files were updated"
+          fi
+      - name: Invoke build workflow
+        uses: benc-uk/workflow-dispatch@4c044c1613fabbe5250deadc65452d54c4ad4fc7
```

Comment:
Instead of using a workflow dispatch here, we _could_ use the builtin `workflow_run`.
See docs here: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onworkflow_runbranchesbranches-ignore
and here: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_run

So the opening part of the build.yml would look like:
```
on:
  workflow_run:
    workflows: [Generate Static Assets]
    types:
      - completed
```

Reply:
Nevermind, this workflow would have to be on the default branch!

Reply:
We tried this 😢 but unfortunately it only triggers on the default branch and we want this to run on commits to the release branches: